### DISCUSSION TITLE
Porting Webkitgtk

### DISCUSSION THREAD

**Discussion by LecrisUT** [Homebrew/discussions#1119] [Tap maintenance and Homebrew development] [answered]
I am in the process of porting webkit's GTK port and so far I have a few questions:
- Are licenses of LGPLv2+ good to port?
- How do we port apps that requires `systemd`? e.g. `mariadb` also spawn services, but how is that controlled and how to I link it to `cmake`?
- Should I include OpenGL compatibility, and if so which formula is that from?

**Comment by SMillerDev** [comment, marked as answer]
I'm assuming you mean that you want to write a formula for some webkit based software.

> Are licenses of LGPLv2+ good to port?

https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae explains the license requirements for homebrew-core.

> How do we port apps that requires systemd? e.g. mariadb also spawn services, but how is that controlled and how to I link it to cmake?

Generally apps don't require systemd on macOS and on Linux we just ignore it I think. I'm not sure what cmake has to do with this.

> Should I include OpenGL compatibility, and if so which formula is that from?

OpenGL is deprecated on macOS but for now you can still use that version.

**Reply by LecrisUT** [reply]
I'm coming back to update `pdfpc` and there were a lot of changes I need to attend to, this being one of them.

I see that LGPL is in the `Debian Free Software Guidelines license` so I guess it's no problem there. I wonder why the webkit has not been officially ported to `homebrew-core` yet. I've seen stranded outdated taps implementing it though.

> Generally apps don't require systemd on macOS and on Linux we just ignore it I think. I'm not sure what cmake has to do with this.

I'll need to check the CMake file to see where it's used. If it's only to install the service automatically than it should be possible to make a `brew service` for that. Are there any guides for doing that?

> OpenGL is deprecated on macOS but for now you can still use that version.

I think I'll just disable the requirement for now and see how it goes.
